engine/src/main/java/org/terasology/input/lwjgl/LwjglMouseDevice.java

```diff
@@ -32,62 +33,73 @@
 import java.util.Queue;
 
 /**
+
  */
 public class LwjglMouseDevice implements MouseDevice, PropertyChangeListener {
     private RenderingConfig renderingConfig;
     private float uiScale;
     private boolean mouseGrabbed;
+    private Queue<MouseAction> queue = Lists.newLinkedList();
+
+    private double xpos = 0.0;
+    private double ypos = 0.0;
+
+    private double xposDelta = 0.0;
+    private double yposDelta = 0.0;
 
     public LwjglMouseDevice(Context context) {
         this.renderingConfig = context.get(Config.class).getRendering();
         this.uiScale = this.renderingConfig.getUiScale() / 100f;
         this.renderingConfig.subscribe(RenderingConfig.UI_SCALE, this);
+
+        // GLFW callback
+        long window = GLFW.glfwGetCurrentContext();
+        GLFW.glfwSetCursorPosCallback(window, this::cursorPosCallback);
+        GLFW.glfwSetMouseButtonCallback(window, this::mouseButtonCallback);
+        GLFW.glfwSetScrollCallback(window, this::scrollCallback);
+
     }
 
     @Override
     public Vector2i getPosition() {
-        return new Vector2i(Mouse.getX() / this.uiScale, (Display.getHeight() - Mouse.getY()) / this.uiScale);
+        return new Vector2i((int) (xpos / this.uiScale), (int)  (ypos/ this.uiScale));
     }
 
     @Override
     public Vector2i getDelta() {
-        return new Vector2i(Mouse.getDX(), -Mouse.getDY());
+        Vector2i result = new Vector2i((int) xposDelta, (int) yposDelta);
+        xposDelta = 0.0;
+        yposDelta = 0.0;
+        return result;
     }
 
     @Override
     public boolean isButtonDown(int button) {
-        return Mouse.isButtonDown(button);
+        return false; //FIXME: low priority unused method
```

Comment:
Should this be noted in an issue somewhere just so it is documented that a button down event for mice won't work?